apim/3.x/tests/api/configmap_mongodb_test.yaml

```diff
@@ -0,0 +1,79 @@
+suite: Test Management API default configmap with MongoDB
+templates:
+  - "api/api-configmap.yaml"
+tests:
+  - it: Set common value and uri
+    template: api/api-configmap.yaml
+    set:
+      mongo:
+        sslEnabled: true
+        socketKeepAlive: true
+        uri: mongodb://mongo-mongodb-replicaset:27017/gravitee
+        auth:
+          enabled: true
+          username: username
+          password: password
+          authSource: authSource
+    asserts:
+      - matchRegex:
+          path: data.[gravitee.yml]
+          pattern: " * mongodb:\n
+                     *  sslEnabled: true\n
+                     *  socketKeepAlive: true\n
+                     *  uri: mongodb://mongo-mongodb-replicaset:27017/gravitee"
+  - it: Set common value and servers
+    template: api/api-configmap.yaml
+    set:
+      mongo:
+        sslEnabled: true
+        socketKeepAlive: true
+        servers: |
+          - host: mongo1
+            port: 27017
+          - host: mongo2
+            port: 27017
+        dbname: dbname
+        auth:
+          enabled: true
+          username: username
+          password: password
+          source: authSource
+    asserts:
+      - matchRegex:
+          path: data.[gravitee.yml]
+          pattern: " * mongodb:\n
+                     *   sslEnabled: true\n
+                     *   socketKeepAlive: true\n
+                     *   servers:\n
+                     *     - host: mongo1\n
+                     *       port: 27017\n
+                     *     - host: mongo2\n
+                     *       port: 27017\n
+                     *\n
+                     *   dbname: dbname\n
+                     *   username: username\n
+                     *   password: password\n
+                     *   authSource: authSource"
+  - it: Set common value and servers
+    template: api/api-configmap.yaml
+    set:
+      mongo:
+        sslEnabled: true
+        socketKeepAlive: true
+        rsEnabled: true
+        dbhost: graviteeio-apim-mongodb-replicaset
+        dbname: dbname
+        dbport: 27117
+        connectTimeoutMS: 2000
+        auth:
+          enabled: true
+          username: username
+          password: password
+          source: authSource
+    asserts:
+      - matchRegex:
+          path: data.[gravitee.yml]
+          pattern: " * mongodb:\n
+                     *   sslEnabled: true\n
+                     *   socketKeepAlive: true\n
+                     *   uri: mongodb://username:password@graviteeio-apim-mongodb-replicaset:27117/dbname\\?&replicaSet=rs0&authSource=authSource&connectTimeoutMS=2000"
```

Comment:
Soooo we end up with `uri: mongodb://username:password@graviteeio-apim-mongodb-replicaset:27117/dbname?&replicaSet=rs0&authSource=authSource&connectTimeoutMS=2000`, notice the strange `dbname?&replicaSet`? I'm not sure mongodb is able to parse this 🤔